src/engine/client/audio.cpp

```diff
@@ -328,3 +317,52 @@ void AudioManager::PlayGlobalSound(const std::string &path) {
     global_source_.reset(CreateSource(path));
     global_source_->StartPlaying();
 }
+
+void AudioManager::SilenceSources() {
+    for(auto sound : sources_) {
+        sound->StopPlaying();
+    }
+
+    global_source_->StopPlaying();
+}
+
+void AudioManager::FreeSources() {
+    LogInfo("freeing all sources...\n");
+
+    SilenceSources();
+    sources_.clear();
```

Comment:
Should do global_source_->reset(NULL); before sources_.clear() to avoid it being destroyed twice.